frontend/src/ProfilePageComp/SavedRecordings.jsx

```diff
@@ -27,24 +29,64 @@ function SavedRecordings() {
           const audioBlob = convertToBlob(item.audios.data, "audio/mp4"); //converts each value in data.audios.data into a blob
           return URL.createObjectURL(audioBlob); //creates URL from blob
         });
-        setRecordings(urls);//Updates state variable with recent list of recordings
+        const ids = data.map((item) => {
+          const numID = item.id;
+          return numID;
+        });
+        setRecordings(urls);
+        setRecordingIDs(ids);
       }
     } catch (error) {
       alert("Failed to fetch audios. Please try again.");
     }
   };
 
+  const handleDelete = async (index) => {
+    try {
+      const audioID = recordingIDs[index]
+      const audioURL = recordings[index]
+      await fetch(
+        `http://localhost:3000/users/${user.id}/audios/${audioID}/delete`,
+        { method: "DELETE" }
+      );
+
+      const updatedIDs = recordingIDs.filter(function (ID) {
+        return ID !== audioID
+      })
+
+      const updatedURLs = recordings.filter(function (url) {
+        return url !== audioURL
+      })
+
+      setRecordingIDs(updatedIDs)
+      setRecordings(updatedURLs)
+
+    }
+    catch (error) {
+
+      alert("Failed to delete audio. Please try again.", error);
```

Comment:
I believe adding an error like this will not display anything. However, you probably want to avoid showing this error directly to the user anyway. You could use `console.error` to log error ([ex](https://www.geeksforgeeks.org/difference-between-error-boundaries-try-catch-in-react/#trycatch)) and then show a separate error to the user. You should be wary of what you display to the user to avoid confusing them